# ⏰ : Change the yaml File path or crosscheck the file path
# ie . replace the below code in the data.ymal file for this Plate Detection 😱

names:
  - License_Plate
nc: 1

path: /content/License-Plate-Recognition-2

train: train/images
val: valid/images
test: test/images

roboflow:
  license: CC BY 4.0
  project: license-plate-recognition-rxg4e-xyod6
  url: https://universe.roboflow.com/smarthsrp/license-plate-recognition-rxg4e-xyod6/dataset/2
  version: 2
  workspace: smarthsrp


# Import Packages

In [ ]:
!pip install roboflow ultralytics

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.9/89.9 kB 6.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 66.8/66.8 kB 6.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.9/49.9 MB 22.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 75.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.4/1.4 MB 85.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.2/4.2 MB 130.0 MB/s eta 0:00:00
  Attempting uninstall: opencv-python-headless
    Found existing installation: opencv-python-headless 4.12.0.88
    Uninstalling opencv-python-headless-4.12.0.88:
      Successfully uninstalled opencv-python-headless-4.12.0.88
  Attempting uninstall: idna
    Found existing installation: idna 3.11
    Uninstalling idna-3.11:
      Successfully uninstalled idna-3.11


# Import Library

In [ ]:
from ultralytics import YOLO
import os

Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.


In [ ]:
!nvidia-smi

Mon Nov 24 15:52:52 2025       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.54.15              Driver Version: 550.54.15      CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  Tesla T4                       Off |   00000000:00:04.0 Off |                    0 |
| N/A   68C    P0             34W /   70W |     852MiB /  15360MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+-----

# Import DataSet


In [ ]:
from roboflow import Roboflow

# Initialize Roboflow
rf = Roboflow(api_key="RoboFlow Api")  # replace with your API key


In [ ]:
# Replace with your workspace and project names
workspace_name = "smarthsrp"
project_name = "license-plate-recognition-rxg4e-xyod6"
version_number = 2  # the version you created/forked

project = rf.workspace(workspace_name).project(project_name)
dataset = project.version(version_number).download("yolov8")  # download in YOLOv8 format use same format in yolov10

print("Dataset downloaded to:", dataset.location)


loading Roboflow workspace...
loading Roboflow project...



Extracting Dataset Version Zip to License-Plate-Recognition-2 in yolov8:: 100%|██████████| 34376/34376 [00:07<00:00, 4857.76it/s]


Dataset downloaded to: /content/License-Plate-Recognition-2


In [ ]:
# 4️⃣ Set paths
dataset_path = "/content/License-Plate-Recognition-2"
data_yaml = os.path.join(dataset_path, "data.yaml")
project_folder = "/content/plate_recognition_project"
run_name = "yolov10_plate"

In [ ]:
print("Train images:", len(os.listdir(os.path.join(dataset_path, "train/images"))))
print("Validation images:", len(os.listdir(os.path.join(dataset_path, "valid/images"))))
print("Test images:", len(os.listdir(os.path.join(dataset_path, "test/images"))))

Train images: 14114
Validation images: 2048
Test images: 1020


# Model checkpoint is Saved in My Drive

In [ ]:
from google.colab import drive   # import os is also required
# 1️⃣ Mount Google Drive
drive.mount('/content/drive')

# 2️⃣ Paths
drive_path = "/content/drive/MyDrive/Licence_Plate"
os.makedirs(drive_path, exist_ok=True)

Mounted at /content/drive


# Load Model

In [ ]:
model=YOLO("/content/drive/MyDrive/Licence_Plate/Licence_Plate_1/weights/last.pt") # Path of the the last.pt or the  last epoch checkpoint i.e epoch50

# Train model

In [ ]:
model.train(
    data=data_yaml,
    epochs=100,
    imgsz=640,
    batch=16,
    #augment=True,
    #mosaic=1.0,            # add mosaic now
    #mixup=0.5,             # add mixup now
    #perspective=0.5,
    project=drive_path,
    name="Licence_Plate_1",
    exist_ok=True,
    save_period=5,
    resume=True
)

Ultralytics 8.3.231 🚀 Python-3.12.12 torch-2.9.0+cu126 CUDA:0 (Tesla T4, 15095MiB)
engine/trainer: agnostic_nms=False, amp=True, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/content/License-Plate-Recognition-2/data.yaml, degrees=0.0, deterministic=True, device=None, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=100, erasing=0.4, exist_ok=True, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=/content/drive/MyDrive/Licence_Plate/Licence_Plate_1/weights/last.pt, momentum=0.937, mosaic=1.0, multi_scale=False, name=Licence_Plate_1, nbs=64, nms=False, opset=None, optimize=False

ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7d0dc8ba2480>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.048048, 

# Result of Validation Set
P (Precision) = 0.981 → Very few false positives

R (Recall) = 0.949 → Most objects detected

mAP50 = 0.979 → Mean Average Precision at 50% IoU, excellent

mAP50-95 = 0.72 → Average over stricter IoU thresholds, still decent

# Run test evaluation on test dataset

In [ ]:
model = YOLO("/content/drive/MyDrive/Licence_Plate/Licence_Plate_1/weights/best.pt")

metrics = model.val(
    data=data_yaml,
    split="test",
    imgsz=640
)

print(metrics)

Ultralytics 8.3.231 🚀 Python-3.12.12 torch-2.9.0+cu126 CUDA:0 (Tesla T4, 15095MiB)
YOLOv10s summary (fused): 106 layers, 7,218,387 parameters, 0 gradients, 21.4 GFLOPs
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 14.3±5.7 MB/s, size: 33.6 KB)
val: Scanning /content/License-Plate-Recognition-2/test/labels... 1020 images, 1 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 1020/1020 977.8it/s 1.0s
val: New cache created: /content/License-Plate-Recognition-2/test/labels.cache
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 64/64 4.5it/s 14.4s
                   all       1020       1085       0.99      0.946      0.979      0.728
Speed: 1.5ms preprocess, 8.7ms inference, 0.0ms loss, 0.2ms postprocess per image
Results saved to /content/runs/detect/val
ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.Confu

# ON Test Set
P (Precision) = 0.99 → Very few false positives.

R (Recall) = 0.946 → Most license plates detected.

mAP50 = 0.979 → Mean Average Precision at IoU 0.5, excellent.

mAP50-95 = 0.728 → Average over stricter IoU thresholds, still good.